src/include/gdb_main.h

```diff
@@ -23,7 +23,9 @@
 
 #include "target.h"
 
+#ifndef GDB_PACKET_BUFFER_SIZE
```

Comment:
There shouldn't be a need to guard this define - it is always only defined here and never in the build system.

If this is because of something you want to do externally to change the buffer sizing, please add a comment before this `#ifndef` explaining its usage (when it is used, how, that 1024 is the absolute lower bound due to the BMP remote protocol) so it's documented and we'll know not to remove the logic here for an external platform.